cliquet/views/errors.py

```diff
@@ -67,9 +67,14 @@ def page_not_found(request):
                                    len(request.path)):]
         redirect = '%s%s' % (path, querystring)
         return HTTPTemporaryRedirect(redirect)
-    elif request.path.startswith('/' + request.registry.route_prefix):
+    elif request.path.startswith('/' + request.registry.route_prefix + '/'):
         errno = ERRORS.MISSING_RESOURCE
         error_msg = "The resource you are looking for could not be found."
+    elif trailing_slash_redirection_enabled:
+        querystring = request.url[(request.url.rindex(request.path) +
+                                   len(request.path)):]
+        redirect = '/%s/%s' % (request.registry.route_prefix, querystring)
+        return HTTPTemporaryRedirect(redirect)
 
     response = http_error(httpexceptions.HTTPNotFound(),
```

Comment:
Is it possible to get through all these ifs without setting `errno` or `error_msg`?